### PR TITLE
Added CORS and new API endpoint for fetching programme specfic courses

### PR DESCRIPTION
**Pull request by Satvik200**

### CODE REVIEW

File: apps/programmes/urls.py [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-
-from apps.programmes.views import ProgrammeViewSet
+from apps.programmes.views import ProgrammeViewSet, ProgrammeStructureView
 
 router = DefaultRouter()
 router.register("programmes", ProgrammeViewSet, basename="programme")
 
 urlpatterns = [
     path("", include(router.urls)),
+    path("programmes/<int:programme_id>/structure", ProgrammeStructureView.as_view(), name="programme-structure"),
```
Comment (josemquinta15) [inline apps/programmes/urls.py]: programme_id is uuid not int

Comment (josemquinta15) [reply apps/programmes/urls.py]: shouldn't the url be `programmes/id/courses` tho? it seems that you'd like to get the courses of the programme

File: apps/programmes/views.py [hunk 2]
```diff
@@ -15,3 +18,21 @@ class ProgrammeViewSet(GenericViewSet, ListModelMixin, RetrieveModelMixin):
...
+    permission_classes = [IsAuthenticated]
+
+    def get(self, request, programme_id):
+        try:
+            programme = Programme.objects.get(id=programme_id)
+            serializer = ProgrammeSerializer(programme)
+            programme_data = serializer.data
+
+            programme_courses = ProgrammeCourse.objects.filter(programme=programme)
+            course_data = ProgrammeCourseSerializer(programme_courses, many=True).data
```
Comment (josemquinta15) [inline apps/programmes/views.py:33]: why not use CourseSerializer?

File: apps/programmes/urls.py (hunk 1 above)
Comment (Satvik200) [reply apps/programmes/urls.py]: Yeah I forgot about the UUID change and did that to int only. Correcting it

File: apps/programmes/views.py (hunk 2 above)
Comment (Satvik200) [reply apps/programmes/views.py:33]: CourseSerializer would give a more general view of the course, lacking the context that ProgrammeCourseSerializer provides with programme-specific prerequisite handling.
CourseSerializer serializes the generic Course data without any programme context

Comment (josemquinta15) [reply apps/programmes/views.py:33]: but programmeCourse is just returning Course info

**Pull request closed without merging**